pkg/monitor/controller_test.go

```diff
@@ -0,0 +1,635 @@
+		t.Run(tt.name, func(t *testing.T) {
+			clientset := fakeclientset.NewSimpleClientset()
+			controller := newControllerMonitor(clientset)
+			initController(controller)
+			err := controller.controllerMonitor.syncNode(tt.key)
+			if tt.expectedError != "" {
+				assert.Equal(t, tt.expectedError, err.Error())
+			} else {
+				assert.Nil(t, err)
+				crd, _ := controller.controllerMonitor.client.CrdV1beta1().AntreaAgentInfos().Get(context.TODO(), "TestExternalNode", metav1.GetOptions{})
+				assert.Equal(t, tt.expectedAgentCRD, crd)
+			}
+		})
+	}
+}
+
+func TestDeleteAgentCRD(t *testing.T) {
+	TestAgentCRD := &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "TestAgentCRD",
+			ResourceVersion: "0",
+			Generation:      0,
+		},
+		NetworkPolicyControllerInfo: v1beta1.NetworkPolicyControllerInfo{
+			NetworkPolicyNum:  1,
+			AddressGroupNum:   1,
+			AppliedToGroupNum: 1,
+		},
+	}
+	tc := []struct {
+		name             string
+		agentCRDName     string
+		existingAgentCRD *v1beta1.AntreaAgentInfo
+		prepareReactor   func(clientset *fakeclientset.Clientset)
+		expectedError    string
+	}{
+		{
+			name:             "CRD Agent with given name exists",
+			agentCRDName:     "TestAgentCRD",
+			prepareReactor:   func(clientset *fakeclientset.Clientset) {},
+			existingAgentCRD: TestAgentCRD,
+			expectedError:    "",
+		},
+		{
+			name:           "CRD Agent with given name does not exist-error was created by NewNotFound",
+			agentCRDName:   "TestAgentCRD",
+			prepareReactor: func(clientset *fakeclientset.Clientset) {},
+			expectedError:  "",
+		},
+		{
+			name:         "CRD Agent with given name does not exist-error not created by NewNotFound ",
+			agentCRDName: "TestAgentCRD",
+			prepareReactor: func(clientset *fakeclientset.Clientset) {
+				clientset.PrependReactor("delete", "antreaagentinfos", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1beta1.AntreaAgentInfo{}, errors.New("error deleting agent crd")
+				})
+			},
+			expectedError: "error deleting agent crd",
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			clientset := fakeclientset.NewSimpleClientset()
+			if tt.existingAgentCRD != nil {
+				clientset = fakeclientset.NewSimpleClientset(tt.existingAgentCRD)
+			}
+			tt.prepareReactor(clientset)
+			controller := newControllerMonitor(clientset)
+			initController(controller)
+			err := controller.controllerMonitor.deleteAgentCRD(tt.agentCRDName)
+			if tt.expectedError != "" {
+				assert.Equal(t, tt.expectedError, err.Error())
+			} else {
+				assert.Nil(t, err)
+				_, err = controller.controllerMonitor.client.CrdV1beta1().AntreaAgentInfos().Get(context.TODO(), tt.agentCRDName, metav1.GetOptions{})
+				assert.Equal(t, "antreaagentinfos.crd.antrea.io \"TestAgentCRD\" not found", err.Error())
+			}
+		})
+	}
+}
+
+func TestCreateAgentCRD(t *testing.T) {
+	TestAgentCRD := &v1beta1.AntreaAgentInfo{
```

Comment:
```suggestion
	testAgentCRD := &v1beta1.AntreaAgentInfo{
```
please check other occurrences and fix them.